<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PasswordGeneratorWidget</class>
 <widget class="QWidget" name="PasswordGeneratorWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>505</width>
    <height>345</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string/>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <item>
    <layout class="QGridLayout" name="gridLayout">
     <item row="0" column="0">
      <widget class="QLabel" name="labelNewPassword">
       <property name="text">
        <string>Password:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <property name="sizeConstraint">
        <enum>QLayout::SetNoConstraint</enum>
       </property>
       <item>
        <widget class="PasswordComboBox" name="editNewPassword">
         <property name="editable">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="togglePasswordButton">
         <property name="checkable">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="labelLength">
       <property name="text">
        <string>Length:</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <layout class="QHBoxLayout" name="horizontalLayout_4">
       <item>
        <widget class="QSlider" name="sliderLength">
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>64</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="tickPosition">
          <enum>QSlider::TicksBelow</enum>
         </property>
         <property name="tickInterval">
          <number>8</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="spinBoxLength">
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>999</number>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Character Types</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <layout class="QGridLayout" name="gridLayout_3">
        <item row="1" column="0">
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <layout class="QVBoxLayout" name="verticalLayoutLeft">
            <item>
             <widget class="QCheckBox" name="checkBoxUppers">
              <property name="text">
               <string>Upper Case ( A - Z )</string>
              </property>
              <attribute name="buttonGroup">
               <string notr="true">optionButtons</string>
              </attribute>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="checkBoxLowers">
              <property name="text">
               <string>Lower Case ( a - z )</string>
              </property>
              <attribute name="buttonGroup">
               <string notr="true">optionButtons</string>
              </attribute>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="checkBoxDigits">
              <property name="text">
               <string>Digits ( 0 - 9 )</string>
              </property>
              <attribute name="buttonGroup">
               <string notr="true">optionButtons</string>
              </attribute>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="checkBoxMinus">
              <property name="text">
               <string>Minus ( - )</string>
              </property>
              <attribute name="buttonGroup">
               <string notr="true">optionButtons</string>
              </attribute>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayoutRight">
            <item>
             <widget class="QCheckBox" name="checkBoxUnderlines">
              <property name="text">
               <string>Underline ( _ )</string>
              </property>
              <attribute name="buttonGroup">
               <string notr="true">optionButtons</string>
              </attribute>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="checkBoxSpaces">
              <property name="text">
               <string>Space (  )</string>
              </property>
              <attribute name="buttonGroup">
               <string notr="true">optionButtons</string>
              </attribute>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="checkBoxSpecials">
              <property name="text">
               <string>Special ( ! @ # $ ... )</string>
              </property>
              <attribute name="buttonGroup">
               <string notr="true">optionButtons</string>
              </attribute>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="checkBoxBrackets">
              <property name="text">
               <string>Brackets ( &lt; &gt; { } [ ] ( ) )</string>
              </property>
              <attribute name="buttonGroup">
               <string notr="true">optionButtons</string>
              </attribute>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <widget class="Line" name="line">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="checkBoxExcludeAlike">
        <property name="text">
         <string>Exclude look-alike characters</string>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">optionButtons</string>
        </attribute>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="checkBoxEnsureEvery">
        <property name="text">
         <string>Ensure that the password contains characters from every group</string>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">optionButtons</string>
        </attribute>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <item>
      <spacer name="horizontalSpacer_2">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="buttonApply">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>Accept</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>PasswordComboBox</class>
   <extends>QComboBox</extends>
   <header location="global">gui/PasswordComboBox.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>editNewPassword</tabstop>
  <tabstop>togglePasswordButton</tabstop>
  <tabstop>sliderLength</tabstop>
  <tabstop>spinBoxLength</tabstop>
  <tabstop>checkBoxUppers</tabstop>
  <tabstop>checkBoxLowers</tabstop>
  <tabstop>checkBoxMinus</tabstop>
  <tabstop>checkBoxUnderlines</tabstop>
  <tabstop>checkBoxSpaces</tabstop>
  <tabstop>checkBoxSpecials</tabstop>
  <tabstop>checkBoxBrackets</tabstop>
  <tabstop>checkBoxExcludeAlike</tabstop>
  <tabstop>checkBoxEnsureEvery</tabstop>
  <tabstop>buttonApply</tabstop>
 </tabstops>
 <resources/>
 <connections/>
 <buttongroups>
  <buttongroup name="optionButtons">
   <property name="exclusive">
    <bool>false</bool>
   </property>
  </buttongroup>
 </buttongroups>
</ui>
